app/src/features/rules/modals/MV3MigrationModal/MigrationModal/MigratedRulesLog.jsx

```diff
@@ -0,0 +1,114 @@
+import { useNavigate } from "react-router-dom";
+import { useCallback, useEffect, useState } from "react";
+import { redirectToRuleEditor } from "utils/RedirectionUtils";
+import { StorageService } from "init";
+import { useSelector } from "react-redux";
+import { getAppMode } from "store/selectors";
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+
+import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
+import LINKS from "config/constants/sub/links";
+
+import { Typography } from "antd";
+
+import RuleIcon from "components/common/RuleIcon";
+import { IoWarningOutline } from "@react-icons/all-files/io5/IoWarningOutline";
+import { RxExternalLink } from "@react-icons/all-files/rx/RxExternalLink";
+import { CiMail } from "@react-icons/all-files/ci/CiMail";
+
+const MigratedRules = () => {
+  const currentlyActiveWorkspace = useSelector(getCurrentlyActiveWorkspace);
+
+  const getMigratedRulesFromLocalStorage = useCallback(() => {
+    const migratedRules = localStorage.getItem("migratedToMV3");
+    const localStorageRulesObject = migratedRules ? JSON.parse(migratedRules) : {};
+    const migrationLogs = localStorageRulesObject?.[currentlyActiveWorkspace?.id];
+    if (migrationLogs?.rulesMigrated) return migrationLogs.rulesMigrationLogs;
+    return [];
+  }, [currentlyActiveWorkspace]);
+
+  const migratedRules = getMigratedRulesFromLocalStorage();
```

Comment:
`migratedRules` will be computed on every render either memoize this or create a useEffect which will read the rules from local storage whenever workspace is switched